src/debug-generator/debug-generator.c

```diff
@@ -59,12 +62,26 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
                 if (value) {
                         r = parse_boolean(value);
-                        if (r < 0)
-                                log_error("Failed to parse systemd.debug_shell= argument '%s', ignoring.", value);
-                        else
-                                arg_debug_shell = r;
-                } else
-                        arg_debug_shell = true;
+                        if (r > 0) {
+                                arg_debug_shell = strdup("y");
+                                if (arg_debug_shell == NULL)
```

Comment:
`!arg_debug_shell` everywhere please according to coding style.

But it turns out that you don't need to allocate anything. Please just use a static string here...

Reply:
It turns out that `proc_cmdline_parse()` allocates the items that are then presented as `const char *`... See the comment below.